### PR TITLE
VCDA-344,367: Reset/Deploy a vapp/vm command

### PR DESCRIPTION
**Pull request by sompa**
- Includes SDK methods for resetting and deploying a vapp or VM(s) within a vapp

- Added cli methods for deploying a vApp/VM
  Added cli methods for resetting a vApp/VM
   Command : **vcd vapp reset sompa-vapp vAppTemplate2 vAppTemplate1**

```
Are you sure you want to reset the vApp or VM(s)? [y/N]: y
vappReset: Resetting Virtual Machine vAppTemplate2(a631e3bb-80a0-47c8-b45e-070321f3621c)
vappReset: Resetting Virtual Machine vAppTemplate2(a631e3bb-80a0-47c8-b45e-070321f3621c)
task: f054db3a-f17a-4338-9937-494e295859fd, Reset Virtual Machine vAppTemplate2(a631e3bb-80a0-47c8-b45e-070321f3621c), result: success
vappReset: Resetting Virtual Machine vAppTemplate1(b455d13b-b66d-484a-82bb-f439a5faea0d)
vappReset: Resetting Virtual Machine vAppTemplate1(b455d13b-b66d-484a-82bb-f439a5faea0d)
task: e870fd9a-ff8d-488e-80a1-81e4eb86031e, Reset Virtual Machine vAppTemplate1(b455d13b-b66d-484a-82bb-f439a5faea0d), result: success
```
Command: **vcd vapp deploy sompa-vapp vAppTemplate1  vAppTemplate2 --power-off** 
Will deploy the vApp/VM in partially running state. Please refer docs for additional info.
```
vappDeploy: Starting Virtual Machine vAppTemplate1(b455d13b-b66d-484a-82bb-f439a5faea0d)
vappDeploy: Starting Virtual Machine vAppTemplate1(b455d13b-b66d-484a-82bb-f439a5faea0d)
task: 79bfcea6-1809-4d43-8247-dd60431494a7, Running Virtual Machine vAppTemplate1(b455d13b-b66d-484a-82bb-f439a5faea0d), result: success
vappDeploy: Starting Virtual Machine vAppTemplate2(a631e3bb-80a0-47c8-b45e-070321f3621c)
vappDeploy: Starting Virtual Machine vAppTemplate2(a631e3bb-80a0-47c8-b45e-070321f3621c)
task: c381fc7a-3e6c-443e-a773-20ea69241e54, Running Virtual Machine vAppTemplate2(a631e3bb-80a0-47c8-b45e-070321f3621c), result: success

```

  Unit tests added for the above to SDK methods.
Tests Run:
```
(pyvcloud) malakars-m01:tests malakars$ python -m unittest vcd_vapp_vm.TestVAppVM.test_1002_deploy_vapp
.
----------------------------------------------------------------------
Ran 1 test in 9.402s

OK
(pyvcloud) malakars-m01:tests malakars$ python -m unittest vcd_vapp_vm.TestVAppVM.test_1003_reset_vapp
.
----------------------------------------------------------------------
Ran 1 test in 17.906s

OK
```

- @rocknes @pacogomez @sahithi @anusuyar @hodgesrm

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/vcd-cli/169)

### CODE REVIEW

**Comment by pacogomez** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[vcd_cli/vapp.py, line 110 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-L3F6VzvxzA5zW5k2IDe:-L3F6VzvxzA5zW5k2IDf:b3zpcm2) ([raw file](https://github.com/vmware/vcd-cli/blob/9e20aba6b2f14fcaac26ba3366895b268ca41ea7/vcd_cli/vapp.py#L110)):*
> ```Python
>         vcd vapp deploy vapp1 vm1 vm2
>             Deploy vm1 and vm2 of vapp1.
> \b
> ```

add example of reset and deploy just the vapp, not the vas

---

*[vcd_cli/vapp.py, line 531 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-L3F8ARsDAh7Vi0S9HsF:-L3F8ARsDAh7Vi0S9HsG:bc4relt) ([raw file](https://github.com/vmware/vcd-cli/blob/9e20aba6b2f14fcaac26ba3366895b268ca41ea7/vcd_cli/vapp.py#L531)):*
> ```Python
> @click.option('--power-on/--power-off',
>               is_flag=True,
>               default=True,
> ```

can this be default=None or be omitted?

---

*[vcd_cli/vapp.py, line 536 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-L3F7KPPY9xvG63XzrOp:-L3F7KPPY9xvG63XzrOq:b-fv83x9) ([raw file](https://github.com/vmware/vcd-cli/blob/9e20aba6b2f14fcaac26ba3366895b268ca41ea7/vcd_cli/vapp.py#L536)):*
> ```Python
>                    'if not specified, default is power on'
>               )
> @click.option('-fc',
> ```

short version for options can only have 1 char, use `-f` or no short version at all. default should be None so default and required can be omitted (see my note in pyvcloud)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/169)*

**Comment by sahithi** [comment]
:lgtm: (after addressing existing comments)

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-:-L3FKfQvEVoS36EWHQe_:b-1lrhgr)*

**Comment by sompa** [comment]
*[vcd_cli/vapp.py, line 536 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-L3F7KPPY9xvG63XzrOp:-L3LzhjAEzgyzEXZtFfQ:b-896fix) ([raw file](https://github.com/vmware/vcd-cli/blob/9e20aba6b2f14fcaac26ba3366895b268ca41ea7/vcd_cli/vapp.py#L536)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

short version for options can only have 1 char, use `-f` or no short version at all. default should be None so default and required can be omitted (see my note in pyvcloud)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/169)*

**Comment by sompa** [comment]
*[vcd_cli/vapp.py, line 531 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-L3F8ARsDAh7Vi0S9HsF:-L3LzjcMzW4q_0809KrV:b-896fix) ([raw file](https://github.com/vmware/vcd-cli/blob/9e20aba6b2f14fcaac26ba3366895b268ca41ea7/vcd_cli/vapp.py#L531)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

can this be default=None or be omitted?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/169)*

**Comment by sompa** [comment]
*[vcd_cli/vapp.py, line 110 at r1](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-L3F6VzvxzA5zW5k2IDe:-L3Lzl0rImpZRwcf1eF2:b-896fix) ([raw file](https://github.com/vmware/vcd-cli/blob/9e20aba6b2f14fcaac26ba3366895b268ca41ea7/vcd_cli/vapp.py#L110)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

add example of reset and deploy just the vapp, not the vas
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/169)*

**Comment by sompa** [comment]
Done addressing the review comments. Will submit if there are no other issues with the pull request.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/vcd-cli/169#-:-L3UBjMB-1jK6IMfTEho:b-dt64ml)*
